cmake/kokkos_test_cxx_std.cmake

```diff
@@ -90,31 +86,31 @@ elseif(KOKKOS_CXX_STANDARD STREQUAL "26")
   set(KOKKOS_CXX_INTERMEDIATE_STANDARD "2C")
   set(KOKKOS_ENABLE_CXX26 ON)
 else()
-  message(FATAL_ERROR "Kokkos requires C++17 or newer but requested ${KOKKOS_CXX_STANDARD}!")
+  message(FATAL_ERROR "Kokkos requires C++20 or newer but requested ${KOKKOS_CXX_STANDARD}!")
 endif()
 
-# Enforce that we can compile a simple C++17 program
+# Enforce that we can compile a simple C++20 program
 
 try_compile(
-  CAN_COMPILE_CPP17 ${KOKKOS_TOP_BUILD_DIR}/corner_cases ${KOKKOS_SOURCE_DIR}/cmake/compile_tests/cplusplus17.cpp
-  OUTPUT_VARIABLE ERROR_MESSAGE CXX_STANDARD 17
+  CAN_COMPILE_CPP20 ${KOKKOS_TOP_BUILD_DIR}/corner_cases ${KOKKOS_SOURCE_DIR}/cmake/compile_tests/cplusplus20.cpp
+  OUTPUT_VARIABLE ERROR_MESSAGE CXX_STANDARD 20
 )
-if(NOT CAN_COMPILE_CPP17)
-  unset(CAN_COMPILE_CPP17 CACHE) #make sure CMake always re-runs this
+if(NOT CAN_COMPILE_CPP20)
+  unset(CAN_COMPILE_CPP20 CACHE) #make sure CMake always re-runs this
```

Comment:
why are we unsetting this here and in L105?

Reply:
Not sure but I was not planning to clean that up as part of this PR